### ISSUE TITLE
unpredictable comment behaviour

### ISSUE THREAD

**Issue by dbHmpf** [go-debos/debos#184]
Hi, 

using debos build from the latest sources (and loving it so far, btw).

According to the documentation I can use comments in the yaml files (#).

But I had some weird behaviour

```
# test

{{- $image := or .image "target.tgz" -}}

architecture: armhf
```
works as expected, but

```
# Build 123

{{- $image := or .image "target.tgz" -}}

architecture: armhf
```
does not work:

```
2019/11/29 14:48:46 Recipe '/home/entw/target.yaml':
2019/11/29 14:48:46   Architecture: 
2019/11/29 14:48:46   Actions:
2019/11/29 14:48:46   - Action: unpack, Description: use base tarball as starting point
...
2019/11/29 14:48:46 Recipe file must have 'architecture' property
```

Any idea why that is?

Regards, 

Sebastian

**Comment by sjoerdsimons** [comment]
run `debos --print-recipe` and it should show your the recipe after golang templating; hopefully that should make it clear

**Comment by obbardc** [comment]
```
# comment.

{{- $image := or .image "target.tgz" -}}

architecture: arm64
```

ends up removing all whitespace (including newline) around the templating characters due to the dash characters, ending up being parsed like:-

```
# comment.architecture: arm64
```

from https://golang.org/pkg/text/template/

>  However, to aid in formatting template source code, if an action's left delimiter (by default "{{") is followed immediately by a minus sign and ASCII space character ("{{- "), all trailing white space is trimmed from the immediately preceding text. Similarly, if the right delimiter ("}}") is preceded by a space and minus sign (" -}}"), all leading white space is trimmed from the immediately following text. In these trim markers, the ASCII space must be present; "{{-3}}" parses as an action containing the number -3.

**Comment by sangorrin** [comment]
I had a similar problem, I avoid putting comments before the template variables

**Comment by andhe** [comment]
This is a duplicate of #81 ?!

**Comment by obbardc** [comment]
similar to #81, but the "-" inside the template remove any whitespace from outside the template

**Comment by sangorrin** [comment]
It seems we can use comments like this:
`{{/* hello comment */}}`

**Comment by sjoerdsimons** [comment]
> It seems we can use comments like this:
> `{{/* hello comment */}}`

Yup that's the suggestion from the golang templating rules indeed (https://golang.org/pkg/text/template/#hdr-Actions).

**Comment by obbardc** [comment]
this is by-design. But perhaps we should document more what `-` does inside template vars (e.g.`{{ -- junk -- }}` )
